packages/data/src/user-preferences/use-user-preferences.js

```diff
@@ -39,140 +120,76 @@ export const useUserPreferences = () => {
-					'variations_report_columns',
-					'dashboard_sections',
-					'dashboard_chart_type',
-					'dashboard_chart_interval',
-					'dashboard_leaderboard_rows',
-					'activity_panel_inbox_last_read',
-					'homepage_layout',
-					'homepage_stats',
-					'android_app_banner_dismissed',
-				];
-
-				// Prep valid fields for update.
-				const metaData = mapValues(
-					pick( userPrefs, userDataFields ),
-					JSON.stringify
-				);
-
-				if ( Object.keys( metaData ).length === 0 ) {
-					return {
-						error: new Error(
-							'No valid woocommerce_meta keys were provided for update.'
-						),
-						updatedUser: undefined,
-					};
-				}
+	const userData = useSelect( ( select ) => {
+		const {
+			getCurrentUser,
+			getEntity,
+			getEntityRecord,
+			getLastEntitySaveError,
+			hasStartedResolution,
+			hasFinishedResolution,
+		} = select( STORE_NAME );
+
+		// Use getCurrentUser() to get WooCommerce meta values.
+		const user = getCurrentUser();
+
+		return {
+			isRequesting:
+				hasStartedResolution( 'getCurrentUser' ) &&
+				! hasFinishedResolution( 'getCurrentUser' ),
+			user,
+			getEntity,
+			getEntityRecord,
+			getLastEntitySaveError,
+		};
+	} );
 
-				// Optimistically propagate new woocommerce_meta to the store for instant update.
-				receiveCurrentUser( {
-					...user,
-					woocommerce_meta: {
-						...user.woocommerce_meta,
-						...metaData,
-					},
-				} );
-
-				// Use saveUser() to update WooCommerce meta values.
-				const updatedUser = await saveUser( {
-					id: user.id,
-					woocommerce_meta: metaData,
-				} );
-
-				if ( undefined === updatedUser ) {
-					// Return the encountered error to the caller.
-					const error = getLastEntitySaveError(
-						'root',
-						'user',
-						user.id
-					);
-
-					return {
-						error,
-						updatedUser,
-					};
+	const updateUserPreferences = ( userPrefs ) => {
+		// WP 5.3.x doesn't have the User entity defined.
+		if ( typeof saveUser !== 'function' ) {
+			// Polyfill saveUser() - wrapper of saveEntityRecord.
+			saveUser = async ( userToSave ) => {
+				const entityDefined = Boolean(
+					userData.getEntity( 'root', 'user' )
+				);
+				if ( ! entityDefined ) {
+					// Add the User entity so saveEntityRecord works.
+					await addEntities( [
+						{
+							name: 'user',
+							kind: 'root',
+							baseURL: '/wp/v2/users',
+							plural: 'users',
+						},
+					] );
 				}
 
-				// Decode the WooCommerce meta after save.
-				const updatedUserResponse = {
-					...updatedUser,
-					woocommerce_meta: getWooCommerceMeta( updatedUser ),
-				};
+				// Fire off the save action.
+				await saveEntityRecord( 'root', 'user', userToSave );
 
-				return {
-					updatedUser: updatedUserResponse,
-				};
-			};
-
-			return {
-				isRequesting:
-					hasStartedResolution( 'getCurrentUser' ) &&
-					! hasFinishedResolution( 'getCurrentUser' ),
-				userPreferences: userData,
-				updateUserPreferences: updateUserPrefs,
```

Comment:
`userData` and `updateUserPrefs` where causing the force re-render as both of them where newly created within the `useSelect`. I moved both outside of the `useSelect`, this required some re-arranging.

The functions like `getEntity`, `getEntityRecord`, `getLastEntitySaveError` aren't a problem, as they are coming from the select method and are constant.